deepspeed/inference/engine.py

```diff
@@ -213,18 +204,16 @@ def forward(self, *inputs, **kwargs):
                 for input in inputs:
                     if torch.is_tensor(input):
                         input = input.to(torch.cuda.current_device())
-                        if self.mp_world_size > 1:
-                            if not input.is_contiguous():
-                                input = input.contiguous()
-                            dist.broadcast(input, 0)
-
+                        if not input.is_contiguous():
+                            input = input.contiguous()
                 for k in kwargs:
                     if torch.is_tensor(kwargs[k]):
                         kwargs[k] = kwargs[k].to(torch.cuda.current_device())
-                        if self.mp_world_size > 1:
-                            if not kwargs[k].is_contiguous():
-                                kwargs[k] = kwargs[k].contiguous()
-                            dist.broadcast(kwargs[k], 0)
+                        if not kwargs[k].is_contiguous():
+                            kwargs[k] = kwargs[k].contiguous()
+                        dist.broadcast(kwargs[k], 0)
```

Comment:
Why didn't you remove this line too?
I think we don't need to use broadcast operation.

Reply:
similar with https://github.com/NVIDIA/Megatron-LM/blob/68797d90935354cf8510dd84711ab6ef5db095f1/megatron/mpu/mappings.py#L85

Reply:
We need to broadcast the inputs all across the model-parallel ranks since after the first token being generated, they might not be synched. You can check this, by removing the broadcasts and run the GPT-Neo test, you will see that the two GPUs print different results. I think this is different than the example you are bringing from the Megatron, they use it mainly to inject all-reduce in the backward.